### ISSUE TITLE
`XMLElement.text` will never be nil.

### ISSUE THREAD

**Issue by ainopara** [drmohundro/SWXMLHash#132]
When implementing #131 , I found that the type of `XMLElement.text` can be changed to `String` from `String?` without modify the implementing.
That is confirmed by making change like this and it is compiled successfully: https://github.com/ainopara/SWXMLHash/commit/8e225b79e1e81d0d0d27787ba078aca3d5e733b3

**Comment by ainopara** [comment]
As a result the last parts of description implementation of XMLElement is equilvalent of 
```
        if true {
            return "<\(name)\(attributesString)>\(text)</\(name)>"
        } else {
            return "<\(name)\(attributesString)/>" // WARN: Will never be executed
        }
```

**Comment by drmohundro** [comment]
Good catch! It looks like this will result in a breaking change (anyone using the library will have to change from `element.text!` to just `element.text`), so it probably best belongs in the next major version (see #84).

I just pushed [version-4.0-changes](https://github.com/drmohundro/SWXMLHash/tree/version-4.0-changes) branch to begin tracking this change. I'd like to get some of the other features from #84 pulled in before I release the next major release.

Thanks again for noticing this!
